src/App.vue

```diff
@@ -450,77 +431,60 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
-        // If Summary Mode: Check validity, save and file else move into summary mode.
-        if (this.isSummaryMode) await this.validateApp()
-        else await this.validateComponents()
+        if (this.isSummaryMode) {
+          // Check validity, and if OK then save and file.
+          await this.validateApp()
+        } else {
+          // Check validity, and if OK then go to summary page.
+          await this.validateCompanyInfoPage()
+        }
         break
     }
   }
 
   /** Redirects to Manage Businesses dashboard. */
   private goToManageBusinessDashboard (): void {
     this.fileAndPayInvalidNameRequestDialog = false
+    this.setHaveUnsavedChanges(false)
+    // FUTURE: Manage Businesses URL should come from config
     const manageBusinessUrl = `${sessionStorage.getItem('AUTH_URL')}business`
-    this.setHaveChanges(false)
     window.location.assign(manageBusinessUrl)
   }
 
   /** Redirects to entity dashboard. */
-  private goToDashboard (force: boolean = false): void {
-    // check if there are no data changes
-    if (!this.hasAlterationChanges || force) {
-      // redirect to dashboard
-      const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
-      window.location.assign(dashboardUrl + this.getBusinessId)
-      return
-    }
+  private goToDashboard (): void {
+    // this.setHaveUnsavedChanges(false)
+    const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
+    window.location.assign(dashboardUrl + this.getBusinessId)
+  }
 
-    // open confirmation dialog and wait for response
-    this.$refs.confirm.open(
-      'Cancel Filings',
-      'Any changes to your company information requiring a fee will be cancelled.',
-      {
-        width: '45rem',
-        persistent: true,
-        yes: 'Cancel my Changes',
-        no: null,
-        cancel: 'Keep my Changes'
-      }
-    ).then(async () => {
-      // Delete the draft filing
-      if (this.getFilingId) {
-        await this.deleteFilingById(this.getFilingId)
-          .then(() => {
-            // redirect to dashboard
-            const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
-            window.location.assign(dashboardUrl + this.getBusinessId)
-          })
-          .catch((error) => {
-            this.$root.$emit('delete-error-event', error)
-          })
-      } else {
-        // redirect to dashboard
-        const dashboardUrl = sessionStorage.getItem('DASHBOARD_URL')
-        window.location.assign(dashboardUrl + this.getBusinessId)
-      }
-    }).catch(() => {
-      // if we get here, No was clicked
-      // nothing to do
-    })
+  private async doDeleteAll (): Promise<void> {
+    // Restore baseline data to original snapshot.
+    this.parseBusinessSnapshot()
+    this.setHaveUnsavedChanges(false)
+    if (this.isSummaryMode) {
+      // just close the Delete All dialog
+      this.confirmDeleteAllDialog = false
+    } else {
+      // redirect to entity dashboard
+      this.goToDashboard()
+    }
```

Comment:
Nice clean up! 
Do we need to restore the baseline if we returning to the dashboard?

Reply:
I believe so, so the "unsaved changes" check doesn't trigger. It also seems like good practice :)

Reply:
Sounds good!